### PR TITLE
Lab rock

### PR DESCRIPTION
**Pull request by andLaing**
Implements a simplified geometry for the walls of HallA at LSC and the option to invoke them for Next-100 and NEW.

### CODE REVIEW

**Comment by andLaing** [comment]
Changed the names to be more specific HallA --> LSCHallA

**Comment by andLaing** [comment]
Happy new year!

What sort of time scale is there for this PR being reviewed? I'd like it done asap so I can integrate the changes into PR #54

**Comment by paolafer** [comment]
> Happy new year!
> 
> What sort of time scale is there for this PR being reviewed? I'd like it done asap so I can integrate the changes into PR #54

It's on the priority list, after the merging and tag of the DEMO geometry.

File: source/geometries/LSCHallA.cc [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  G4ThreeVector LSCHallA::GenerateVertex(const G4String& region) const
+  {
+    G4ThreeVector vertex(0., 0., 0.);
+    if (region == "HALLA_INNER")
+      return hallA_vertex_gen_->GenerateVertex("INNER_SURFACE");
+    else if (region == "HALLA_OUTER")
+      return hallA_outer_gen_->GenerateVertex("INNER_SURFACE");
+    // Unknown region
+    else {
+      G4Exception("[HALLA]", "GenerateVertex()", FatalException,
```
Comment (paolafer) [inline source/geometries/LSCHallA.cc]: Please, update the exception error following the [coding conventions](https://github.com/next-exp/nexus/wiki/Coding-conventions-and-style-guide).

File: source/geometries/LSCHallA.cc [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+#include <G4NistManager.hh>
+#include <G4ProductionCuts.hh>
+#include <G4PVPlacement.hh>
+#include <G4Region.hh>
+#include <G4SubtractionSolid.hh>
+#include <G4Tubs.hh>
+#include <G4VisAttributes.hh>
+#include <Randomize.hh>
+
+#include <CLHEP/Units/PhysicalConstants.h>
```
Comment (paolafer) [inline source/geometries/LSCHallA.cc]: These two includes are not necessary anymore.

Comment (paolafer) [reply source/geometries/LSCHallA.cc]: Review all the includes to eliminate those that are not used.

**Comment by paolafer** [comment]
I would add a visibility for the rock. It can be turned on and off by parameter, if needed.

File: macros/NEW_hallA.init.mac [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+/PhysicsList/RegisterPhysics G4DecayPhysics
+/PhysicsList/RegisterPhysics G4RadioactiveDecayPhysics
+/PhysicsList/RegisterPhysics NexusPhysics
+/PhysicsList/RegisterPhysics G4StepLimiterPhysics
+
+/Geometry/RegisterGeometry NEXT_NEW
+
+/Generator/RegisterGenerator SINGLE_PARTICLE
+
+/Actions/RegisterTrackingAction VALIDATION
```
Comment (paolafer) [inline macros/NEW_hallA.init.mac]: There's no need to use the `VALIDATION` tracking action here. Use `DEFAULT` instead.

File: source/geometries/Next100.cc [hunk 4]
```diff
@@ -140,8 +162,18 @@ namespace nexus {
...
     ics_->Construct();
 
-    new G4PVPlacement(0, G4ThreeVector(0., 0., -gate_zpos_in_vessel_), shielding_logic,
-     		      "LEAD_BOX", lab_logic_, false, 0);
+    if (lab_walls_){
+      G4ThreeVector castle_pos(0., hallA_walls_->GetLSCHallACastleY(),
+			       hallA_walls_->GetLSCHallACastleZ());
+      new G4PVPlacement(0, castle_pos, shielding_logic, "LEAD_BOX",
+       			hallA_logic_, false, 0);
+      G4ThreeVector gate_pos(0., 0., -gate_zpos_in_vessel_);
```
Comment (paolafer) [inline source/geometries/Next100.cc]: This variable could be defined outside the loop and used in both cases.

**Review by paolafer — APPROVED** [review]
This PR implements the possibility of adding the LSC rock in the hall A to the simulation. Some example macros are added to show how to use it both in the NEW and in the NEXT100 geometry. I approve it.

**Pull request merged**